crates/cairo-lint-core/src/lints/bool_comparison.rs

```diff
@@ -0,0 +1,30 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::kind::SyntaxKind;
+use cairo_lang_syntax::node::SyntaxNode;
+
+pub const BOOL_COMPARISON: &str = "Unnecessary comparison with a boolean value. Use the variable directly.";
+
+pub fn check_bool_comparison(db: &dyn SyntaxGroup, node: SyntaxNode, diagnostics: &mut Vec<PluginDiagnostic>) {
+    print!("check_bool_comparison\n");
+    if let SyntaxKind::ExprBinary = node.kind(db) {
```

Comment:
Now that you are working with `ExprBinary` you can extract the `lhs`,  `rhs` and `op`. It'll avoid relying on the text representation.
You can check [that](https://github.com/keep-starknet-strange/cairo-lint/blob/0abf20e3b5cb0bebdd1d8efb39ae5f8eb1f4b1c6/crates/cairo-lint-core/src/lints/operator.rs#L9-L15) it could help you

Reply:
got it! I updated it to use those